### DISCUSSION TITLE
Experimental curl features delivery ideas

### DISCUSSION THREAD

**Discussion by talregev** [curl/curl#9489] [Ideas]
@bagder 
I just saw your YouTube of curl up with experimental feature inside curl:
https://youtu.be/0urDlSaKptI?t=418 
I have an idea for how people can easily build curl with experimental feature.
You can contribute for vcpkg. Many project compiled curl with vcpkg, that already enable some of the features.
So simple way is to contribute vcpkg curl port that enable the experimental features.
https://github.com/microsoft/vcpkg
https://github.com/microsoft/vcpkg/tree/master/ports/curl
![image](https://user-images.githubusercontent.com/7043539/189708441-9e324fb0-49d4-4fc5-b823-37785c86b6eb.png)

Another idea is to compile curl with all the experimental features,  and enable them with flags.

**Comment by bagder** [comment]
> You can contribute for vcpkg

I don't think that would help much. People simply don't build curl with experimental features enabled, and frankly people on Windows in general build curl even less than people on other platforms (and I assume vcpkg is mostly Windows people).

> Another idea is to compile curl with all the experimental features, and enable them with flags

Yeah, maybe. But how do we avoid that users then just put those options into production and then get very sad and upset if we change them, as we want to allow ourselves to do with experimental features?

**Reply by talregev** [reply]
> > You can contribute for vcpkg
> 
> I don't think that would help much. People simply don't build curl with experimental features enabled, and frankly people on Windows in general build curl even less than people on other platforms (and I assume vcpkg is mostly Windows people).
> 
> > Another idea is to compile curl with all the experimental features, and enable them with flags
> 
> Yeah, maybe. But how do we avoid that users then just put those options into production and then get very sad and upset if we change them, as we want to allow ourselves to do with experimental features?

vcpkg is for all platforms. On the opensource I contribute we compiled curl with vcpkg for linux. windows, mac and android. also in arm, and arm64 versions.
But it also can compiled in ios and other triplets.
And it really help when you have a simple build tool. 
If http3 and WebSockets will be enable on vcpkg, I can test it and report back.

**Reply by bagder** [reply]
>  If http3 and WebSockets will be enable on vcpkg, I can test it and report back.

I'm sure someone fluent in vcpkg is able to add that ability to their build.

**Reply by talregev** [reply]
> Yeah, maybe. But how do we avoid that users then just put those options into production and then get very sad and upset if we change them, as we want to allow ourselves to do with experimental features?

There always be people that will get upset that you will change something in curl, even deprecated functionality that no one is use. 
You understand that the current situation is no one is using experimental features in curl and needed to make simpler for users.
So, if vcpkg no, and compile all the experimental features is no. Do you really looking for a new way to ship the experimental features?

**Reply by bagder** [reply]
I thought this was a *discussion* ?

I haven't said not to vcpkg. Why would I do that? I would welcome better vcpkg support - even if I don't even know what it can do today since I have no knowledge about it.

I *asked a question* about built experimental features. A question I think is very valid and important, and one that I have no good answer to myself.

**Comment by talregev** [comment]
The answer for the experimental feature that is changing, is the same answer for deprecated stuff you remove  from curl.
You learn yourself now what is best for curl. But for general there is a tik tok approach:
- version is api created
- version api announced deprecated.
- version api removed / change.
